backend/lcfs/web/api/compliance_report/repo.py

```diff
@@ -541,32 +551,40 @@ async def add_compliance_report_summary(
 
     @repo_handler
     async def save_compliance_report_summary(
-        self, summary_id: int, summary: ComplianceReportSummarySchema
+        self, report_id: int, summary: ComplianceReportSummarySchema
     ):
         """
         Save the compliance report summary to the database.
 
-        :param summary_id: The ID of the compliance report summary
+        :param report_id: The ID of the compliance report
         :param summary: The generated summary data
         """
-        existing_summary = await self.get_summary_by_id(summary_id)
+        existing_summary = await self.get_summary_by_report_id(report_id)
 
         if existing_summary:
             summary_obj = existing_summary
         else:
-            raise ValueError(f"No summary found with ID {summary_id}")
+            raise ValueError(f"No summary found with report ID {report_id}")
 
         # Update renewable fuel target summary
         for row in summary.renewable_fuel_target_summary:
             line_number = row.line
             for fuel_type in ["gasoline", "diesel", "jet_fuel"]:
                 column_name = f"line_{line_number}_{row.field.lower()}_{fuel_type}"
-                setattr(summary_obj, column_name, int(getattr(row, fuel_type) or 0))
+                setattr(
```

Comment:
Main change of this MR, these would before default to 0. Now we copy over the users previously entered value and then the calculate function will reset it to the max if the max has changed.